src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,171 @@
+        user.setLogin(null);
+        assertThrows(InvalidUserInputException.class,
+                () -> registrationService.register(user), INPUT_EXCEPTION_MESSAGE);
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user.setPassword(null);
+        assertThrows(InvalidUserInputException.class,
+                () -> registrationService.register(user), INPUT_EXCEPTION_MESSAGE);
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user.setAge(null);
+        assertThrows(InvalidUserInputException.class,
+                () -> registrationService.register(user), INPUT_EXCEPTION_MESSAGE);
+    }
+
+    @Test
+    void register_smallAge_notOk() {
+        user.setAge(17);
+        assertThrows(InvalidUserInputException.class,
+                () -> registrationService.register(user), INPUT_EXCEPTION_MESSAGE);
+    }
+
+    @Test
+    void register_highAge_notOk() {
+        user.setAge(160);
+        assertThrows(InvalidUserInputException.class,
+                () -> registrationService.register(user), INPUT_EXCEPTION_MESSAGE);
+    }
+
+    @Test
+    void register_minusAge_notOk() {
+        user.setAge(-50);
+        assertThrows(InvalidUserInputException.class,
+                () -> registrationService.register(user), INPUT_EXCEPTION_MESSAGE);
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        user.setLogin("o");
+        assertThrows(InvalidUserInputException.class,
+                () -> registrationService.register(user), INPUT_EXCEPTION_MESSAGE);
+    }
+
+    @Test
+    void register_longLogin_notOk() {
+        user.setLogin("Ilko12345678901234567890");
+        assertThrows(InvalidUserInputException.class,
+                () -> registrationService.register(user), INPUT_EXCEPTION_MESSAGE);
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        user.setPassword("12345");
+        assertThrows(InvalidUserInputException.class,
+                () -> registrationService.register(user), INPUT_EXCEPTION_MESSAGE);
+    }
+
+    @Test
+    void register_longPassword_notOk() {
+        user.setPassword("1234567890QWERTY_qwerty");
+        assertThrows(InvalidUserInputException.class,
+                () -> registrationService.register(user), INPUT_EXCEPTION_MESSAGE);
+    }
+
+    @Test
+    void register_usersInSequence_ok() {
+        assertEquals(1, (long) registrationService.register(user).getId(),
+                USER_RETURN_EXCEPTION_MESSAGE);
+
+        user = new User();
+        user.setLogin("GOOD");
+        user.setPassword(CORRECT_PASSWORD);
+        user.setAge(CORRECT_AGE);
+        assertEquals(2, (long) registrationService.register(user).getId(),
+                USER_RETURN_EXCEPTION_MESSAGE);
+
+        user = new User();
+        user.setLogin("User_3");
+        user.setPassword("Some_Password");
+        user.setAge(55);
+        assertEquals(3, (long) registrationService.register(user).getId(),
+                USER_RETURN_EXCEPTION_MESSAGE);
```

Comment:
let's also check if storage size increases

Reply:
umm, it definitely is because of id comparing... I'm not sure what you mean